flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/serialization/SpanningRecordSerializationTest.java

```diff
@@ -104,11 +116,216 @@ public void testHandleMixedLargeRecords() throws Exception {
 		testSerializationRoundTrip(originalRecords, segmentSize);
 	}
 
+	/**
+	 * Non-spanning, deserialization reads one byte too many and succeeds.
+	 */
+	@Test
+	public void testHandleDeserializingTooMuchNonSpanning1() throws Exception {
+		expectedException.expect(IOException.class);
+		expectedException.expectMessage(" -1 remaining unread byte");
+		testHandleWrongDeserialization(new StringValueDeserializingTooMuch("Test string"), 32 * 1024);
+	}
+
+	/**
+	 * Non-spanning, serialization length is 16 (including headers), deserialization reads one byte
+	 * too many and succeeds.
+	 */
+	@Test
+	public void testHandleDeserializingTooMuchNonSpanning2() throws Exception {
+		expectedException.expect(IOException.class);
+		expectedException.expectMessage(" -1 remaining unread byte");
+		testHandleWrongDeserialization(new StringValueDeserializingTooMuch("Test string"), 17);
```

Comment:
This test can be merged with above `testHandleDeserializingTooMuchNonSpanning1()` by adding `testHandleWrongDeserialization(new StringValueDeserializingTooMuch("Test string"), 17)` at last? Because the `expectedException` is same.

Reply:
Actually, I'd like to follow a one-scenario-per-test pattern for better isolation and testing specific things during debugging etc.
Let me try to de-dup in another way: including the `ExpectedException` setup into `testHandleWrongDeserialization`

Reply:
👍